api/http/proxy/factory/docker/build.go

```diff
@@ -23,29 +25,81 @@ type postDockerfileRequest struct {
 // In any other case, it will leave the request unaltered.
 func buildOperation(request *http.Request) error {
 	contentTypeHeader := request.Header.Get("Content-Type")
-	if contentTypeHeader != "" && !strings.Contains(contentTypeHeader, "application/json") {
-		return nil
-	}
 
-	var dockerfileContent []byte
+	mediaType, _, err := mime.ParseMediaType(contentTypeHeader)
+	if err != nil {
+		return err
+	}
 
-	if contentTypeHeader == "" {
+	var buffer []byte
+	switch mediaType {
+	case "":
 		body, err := ioutil.ReadAll(request.Body)
 		if err != nil {
 			return err
 		}
-		dockerfileContent = body
-	} else {
+
+		buffer, err = archive.TarFileInBuffer(body, "Dockerfile", 0600)
+		if err != nil {
+			return err
+		}
+
+	case "application/json":
 		var req postDockerfileRequest
 		if err := json.NewDecoder(request.Body).Decode(&req); err != nil {
 			return err
 		}
-		dockerfileContent = []byte(req.Content)
-	}
 
-	buffer, err := archive.TarFileInBuffer(dockerfileContent, "Dockerfile", 0600)
-	if err != nil {
-		return err
+		buffer, err = archive.TarFileInBuffer([]byte(req.Content), "Dockerfile", 0600)
+		if err != nil {
+			return err
+		}
+
+	case "multipart/form-data":
+		err := request.ParseMultipartForm(32 << 20) // 32 MB
```

Comment:
Do you think the following form might be more readable?
```
const OneMegabyte = 1024768
...
err := request.ParseMultipartForm(32 * OneMegabyte) // limit parser memory to 32MB
```

Reply:
Sure. Fixed